lib/srv/db/common/errors.go

```diff
@@ -102,3 +104,57 @@ type causer interface {
 type pgError interface {
 	Unwrap() error
 }
+
+// ConvertConnectError converts common connection errors to trace errors with
+// extra information/recommendations if necessary.
+func ConvertConnectError(err error, sessionCtx *Session) error {
+	errString := err.Error()
+	switch {
+	case strings.Contains(errString, "x509: certificate signed by unknown authority"):
+		return trace.AccessDenied("Database service cannot validate database's certificate: %v. Please verify if the correct CA bundle is used in the database config.", err)
```

Comment:
I assume we don't mind if the errors get wrapped with additional messages along the way, as long as this one gets shown as well?